File: PCS/paperroute.py
S = int(input())
for _ in range(S):
    H = int(input())
    P = [int(x) for x in input().split()]
    maxp = 0
    sum = 0
    for p in P:
        if p < 0:
            if sum > maxp:
                maxp = sum
        if sum + p < 0:
            sum = 0
        else:
            sum += p
    maxp = max(sum, maxp)
    print(maxp)

File: PCS/paperroute.cpp
#include <iostream>

using namespace std;

int main() {
    ios::sync_with_stdio(false); cin.tie(0);
    int S;
    cin >> S;
    for(int i = 0; i < S; ++i) {
        int H;
        cin >> H;
        int maxp = 0;
        int sum = 0;
        while(H--) {
            int p;
            cin >> p;
            if(p < 0) {
                if(sum > maxp) {
                    maxp = sum;
                }
            }
            if(sum + p < 0) {
                sum = 0;
            } else {
                sum += p;
            }
        }
        maxp = maxp > sum ? maxp : sum;
        cout << maxp << '\n';
    }
    return 0;
}